zjit/src/hir.rs

```diff
@@ -680,6 +680,9 @@ impl Function {
         } else {
             self.blocks[block.0].insns.push(id);
         }
+        if self.insns[id.0].has_output() {
+            self.insn_types[id.0] = types::Any;
```

Comment:
Is it possible to do this at `new_insn()` by checking `insn.has_output()`?

Reply:
Yeah. I'm trying to figure out what makes me want to do it here and what the differences are between an instruction that's been added to a block vs not

Reply:
I'm worried about a possible situation that we initialize it with `types::Empty` at `new_insn()`, we update it to something before pushing the instruction, and then clear it with `types::Any` here. I'm thinking it's safer to set the default value first, not later like this.

Reply:
I agree with you and changed it. I think you are right